code-challenges.test.js

```diff
@@ -1,46 +1,145 @@
+  it("takes in an array of objects and returns an array with a sentence about each person with their name capitalized", () => {
+    const people = [
+      { name: "ford prefect", occupation: "a hitchhiker" },
+      { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
+      { name: "arthur dent", occupation: "a radio employee" }
+    ]
+    const peopleOutput = ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
+    expect(nameCapitalizer(people)).toEqual(peopleOutput)
+  })
+})
+
+  // $ yarn jest (results)
+  // ReferenceError: nameCapitalizer is not defined
+  // Test Suites: 1 failed, 1 total
+  // Tests:       1 failed, 1 total
 
 // b) Create the function that makes the test pass.
 
-// --------------------2) Create a function that takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
+  // Create nameCapitalizer function with const keyword. Pass in (arr) for its parameter.
+  // Map over the array and return its result. Pass through (obj) as the parameter for .map()
+  // Bear with me in explaining the huge chain of methods. I will explain it in logical steps to achieve each specific task. Everything is written with backticks and string interpolation.
+    // CAPITALIZE FIRST LETTER OF FIRST NAME.
+      // obj.name.charAt(0).toUpperCase()
+        // obj.name pulls the entire name from each object ( "ford prefect"/"zaphod beeblebrox"/"arthur dent" ).
+        // Append obj.name with .charAt(0).toUpperCase() to capitalize the first letter of first name.
+    // LOWERCASE THE REST OF THE FIRST NAME.
+      // obj.name.slice(1, obj.name.indexOf(' '))
+        // Append obj.name with .slice() and specify the starting/ending arguments. Starts with 1 (index of second character) and ends with obj.name.indexOf(' ').
+        // This tells the .slice() method to stop at the index of the empty character (the space between first and last name).
+    // CAPITALIZE FIRST LETTER OF LAST NAME.
+      // obj.name.slice(obj.name.indexOf(' ')).charAt(1).toUpperCase()
+        // Slice obj.name starting with .indexOf(' '), which is the index of the empty character. This essentially pulls JUST the last name.
+        // Append THAT with .charAt(1).toUpperCase(). This will uppercase the first letter of the last name.
+    // LOWERCASE THE REST OF THE LAST NAME.
+      // obj.name.slice(obj.name.indexOf(' ')).slice(2)
+        // Pull just the last name once more with obj.name.slice(obj.name.indexOf(' '))
+        // From there, we can call .slice() once more to get the rest of the last name after the first letter.
+
+const nameCapitalizer = (arr) => {
+  return arr.map(obj => `${obj.name.charAt(0).toUpperCase()}${obj.name.slice(1, obj.name.indexOf(' '))} ${obj.name.slice(obj.name.indexOf(' ')).charAt(1).toUpperCase()}${obj.name.slice(obj.name.indexOf(' ')).slice(2)} is ${obj.occupation}.`)
+}
```

Comment:
This is awesome! You can make this more dynamic by utilizing Object.keys and Object.values